frontend/apps/filemanager/filemanagershortcuts.lua

```diff
@@ -7,199 +7,210 @@ local Screen = require("device").screen
-                            self.ui.file_chooser:changeToPath(folder)
-                        else -- called from Reader
-                            self.ui:onClose()
-                            self.ui:showFileManager(folder .. "/")
-                        end
-                    end
-                end
-            end,
+            name = item.text,
         })
     end
-
     table.sort(item_table, function(l, r)
         return l.text < r.text
     end)
 
     -- try to stay on current page
     local select_number
-
-    if self.fm_bookmark.page and self.fm_bookmark.perpage and self.fm_bookmark.page > 0 then
-        select_number = (self.fm_bookmark.page - 1) * self.fm_bookmark.perpage + 1
+    if self.shortcuts_menu.page and self.shortcuts_menu.perpage and self.shortcuts_menu.page > 0 then
+        select_number = (self.shortcuts_menu.page - 1) * self.shortcuts_menu.perpage + 1
     end
-
-    self.fm_bookmark:switchItemTable(nil,
-                                     item_table, select_number)
+    self.shortcuts_menu:switchItemTable(nil, item_table, select_number)
 end
 
-function FileManagerShortcuts:addNewFolder()
-    local PathChooser = require("ui/widget/pathchooser")
-    local path_chooser = PathChooser:new{
-        select_directory = true,
-        select_file = false,
-        path = self.fm_bookmark.curr_path,
-        onConfirm = function(path)
-            local add_folder_input
-            local friendly_name = util.basename(path) or _("my folder")
-            add_folder_input = InputDialog:new{
-                title = _("Enter friendly name"),
-                input = friendly_name,
-                description = T(_("Title for selected folder:\n%1"), BD.dirpath(path)),
-                buttons = {
-                    {
-                        {
-                            text = _("Cancel"),
-                            id = "close",
-                            callback = function()
-                                UIManager:close(add_folder_input)
-                            end,
-                        },
-                        {
-                            text = _("Add"),
-                            is_enter_default = true,
-                            callback = function()
-                                self:addFolderFromInput(add_folder_input:getInputValue(), path)
-                                UIManager:close(add_folder_input)
-                            end,
-                        },
-                    }
-                },
-            }
-            UIManager:show(add_folder_input)
-            add_folder_input:onShowKeyboard()
+function FileManagerShortcuts:_getIndex(folder)
+    for i, v in ipairs(self.folder_shortcuts) do
+        if v.folder == folder then
+            return i
         end
-    }
-    UIManager:show(path_chooser)
+    end
+end
+
+function FileManagerShortcuts:isShortcutAdded(folder)
+    return self:_getIndex(folder) and true or false
```

Comment:
`:isShortcutAdded(path)` would read better (where it is called elsewhere in this PR) as `:hasFolderShorcut(path)`